app/src/main/java/org/fossasia/pslab/fragment/LAChannelModeFragment.java

```diff
@@ -197,14 +204,55 @@ public void onClick(View v) {
         return selectView;
     }
 
-    public interface OnChannelSelectedListener {
-        void channelSelectedNowAnalyze(Bundle params);
+    @SuppressLint("ResourceType")
+    public void howToConnectDialog(String title, String intro, int iconID, String desc) {
+        try {
+            final AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
+            LayoutInflater inflater = getLayoutInflater();
+            View dialogView = inflater.inflate(R.layout.custom_dialog_box, null);
+            final SharedPreferences settings = getActivity().getSharedPreferences(PREFS_NAME, 0);
+            Boolean skipMessage = settings.getBoolean("skipMessage", false);
+
+            dontShowAgain = (CheckBox) dialogView.findViewById(R.id.toggle_show_again);
+            final TextView heading_text = (TextView) dialogView.findViewById(R.id.custom_dialog_text);
+            final TextView description_text = (TextView) dialogView.findViewById(R.id.description_text);
+            final ImageView schematic = (ImageView) dialogView.findViewById(R.id.custom_dialog_schematic);
+            final Button ok_button = (Button) dialogView.findViewById(R.id.dismiss_button);
+
+            builder.setView(dialogView);
+            builder.setTitle(title);
+            heading_text.setText(intro);
+            schematic.setImageResource(iconID);
+            description_text.setText(desc);
+            final AlertDialog dialog = builder.create();
+            ok_button.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    Boolean checkBoxResult = false;
+                    if (dontShowAgain.isChecked())
+                        checkBoxResult = true;
+                    SharedPreferences settings = getActivity().getSharedPreferences(PREFS_NAME, 0);
```

Comment:
Why is it this SharedPreference initialized again? You can use the previously initialized variable can't you?